Creating a new SEO topic for the doc guide (#1899)
writethedocs/www: docs/guide/seo.rst

================================

> Search Engine Optimization (SEO)

Reviewer: ```suggestion
Search engine optimization (SEO) for documentation
```

Reviewer: optional, but considering the SEO of this page itself, it could do to have a more descriptive title

Author: :+1:

================================

What is search engine optimization?
-----------------------------------

Search engines drive a significant portion of site traffic for docs that publish as public-facing websites.

> Search engine optimization (SEO) describes the means and methods used to have your web pages display for a set of desired search terms.

Reviewer: ```suggestion
Search engine optimization (SEO) describes the methods used to improve the rank of your web pages in search results for a given set of search terms. 
```

Author: Nice rewrite. Thanks.

> The work of SEO demands familiarity with the ever-changing rules of the algorithms search engines use to select and arrange web sites to return for any particular query.

Reviewer: ```suggestion
To perform SEO, you must be familiar with the ever-changing algorithms that search engines use to select and arrange the webpages returned for a given set of search terms.
```

Author: :+1:

These algorithms change frequently, so the requirements for achieving the best optimization also change frequently.

> 
> Fortunately, good tech writing already does most of the standard practices required for excellent SEO.

Reviewer: ```suggestion

SEO for documentation
------------------------

Good tech writing follows most of the standard practices required for excellent SEO.
```

(so that we call out the documentation-specific considerations)

Author: Great suggestion

> - Write good headings

Reviewer: Do we want to be more descriptive about "good"?

- imperative, user-centric, goal-oriented headings?

Author: Sounds good to me. I don't know much about SEO and was just kind of pulling from the Slack thread that sparked this.

- Make your docs skimmable by breaking content into sections under appropriate headings
- Focus on good information architecture that provides quality navigation and site hierarchy

> - Include page metadata

Reviewer: where? and can we provide examples (like last updated date?)

Author: How about
```
- Include page metadata, such as last modified date, in the `<head>` section of your web page (how to do this varies by your publication toolchain)
```

Reviewer: let me be more constructive:

```suggestion
- Make sure your webpages include metadata, such as that described in the `The Ultimate Guide to SEO Meta Tags <https://moz.com/blog/the-ultimate-guide-to-seo-meta-tags>`__ by Moz.
```

Reviewer: Jinx comments! I think your suggestion is excellent!

Author: Gonna drop that Moz link into the blog list, though!

- Work on making sure your documentation meets standards for accessibility

> Resources

Reviewer: ```suggestion
SEO resources
```

Author: :+1:

---------

Search engine documentation
~~~~~~~~~~~~~~~~~~~~~~~~~~~

Review the search engine documentation guides:

- Google's `Search Central <https://developers.google.com/search/docs>`__
- Google's `How Search Works <https://www.google.com/search/howsearchworks/how-search-works/ranking-results/>`_ guide
- Bing's `Webmaster Guidelines <https://www.bing.com/webmasters/help/webmasters-guidelines-30fba23a>`__

> WTD resources on SEO

Reviewer: duplicates the topics page, do we want to link to that page instead of duplicating those links?

Author: I tried in an earlier commit, but couldn't figure out how to link to the specific heading.
I'd rather link there if someone else knows how to get that done.

Reviewer: I think `/topics/#search` should work

Author: I don't see Sphinx interlinking used anywhere else, but gonna give it a shot, as that's what we use in the MinIO docs.

Author: > I think `/topics/#search` should work

You would like to think. Something else must also be broken, but this link is giving me headaches.

~~~~~~~~~~~~~~~~~~~~

- |:movie_camera:| `The Search for Search </videos/portland/2020/the-search-for-search-paul-logan/>`__
- |:movie_camera:| `Making documentation discoverable in search engines </videos/prague/2020/making-documentation-discoverable-in-search-engines-myriam-jessier/>`__

Books
~~~~~

- `SEO for Everyone <https://abookapart.com/products/seo-for-everyone>`__, by Rebekah Baggs & Chris Corak

Blogs and websites
~~~~~~~~~~~~~~~~~~

- Moz `SEO Beginners Guide <https://moz.com/beginners-guide-to-seo>`__
- `Search Engine Land <https://searchengineland.com/>`__
- `Hubspot's list of SEO blogs <https://blog.hubspot.com/marketing/best-seo-blogs>`__

> - https://ahrefs.com/seo

Reviewer: ```suggestion
- `Search Engine Optimization: A beginner’s guide from Ahrefs <https://ahrefs.com/seo>`__
```

Author: :+1:

> - http://backlinko.com/hub/seo

Reviewer: ```suggestion
- `SEO Marketing Hub 2.0 from Backlinko <http://backlinko.com/hub/seo>`__
```

Author: :+1:

Tools
~~~~~

> - `Headline Analyzer <https://coschedule.com/headline-analyzer>`__

Reviewer: ```suggestion
- `Headline Analyzer <https://coschedule.com/headline-analyzer>`__
- `Google Search Console <https://search.google.com/search-console/about>`__
- `Ahrefs Site Audit Tool <https://ahrefs.com/site-audit>`__
- `Keywords Everywhere <https://keywordseverywhere.com/>`__ web browser extension
```

Author: Thanks for the extra links!

==================================================
Search engine optimization (SEO) for documentation
==================================================

What is search engine optimization?
-----------------------------------

Search engines drive a significant portion of site traffic for docs that publish as public-facing websites.
Search engine optimization (SEO) describes the methods used to improve the rank of your web pages in search results for a given set of search terms.

To perform SEO, you must be familiar with the ever-changing algorithms that search engines use to select and arrange the webpages returned for a given set of search terms.
These algorithms change frequently, so the requirements for achieving the best optimization also change frequently.

SEO for documentation
---------------------

Good tech writing follows most of the standard practices required for excellent SEO:

- Write imperative, user-centric, goal-oriented headings
- Make your docs skimmable by breaking content into sections under appropriate headings
- Focus on good information architecture that provides quality navigation and site hierarchy

> - Include page metadata, such as last modified date, in the `<head>` section of your web page (how to do this varies by your publication toolchain)

Reviewer 2: I believe this is the link causing the issue. To avoid thinking this is a link, you need to double escape it. I think
```suggestion
- Include page metadata, such as last modified date, in the ``<head>`` section of your web page (how to do this varies by your publication toolchain)
```

Author: AHHHHHHHHHHHHHHH
Thank you for your eyes.

- Make sure your documentation meets :doc:`standards for accessibility </guide/writing/accessibility>`

SEO resources
-------------

Search engine documentation
~~~~~~~~~~~~~~~~~~~~~~~~~~~

Review the search engine documentation guides:

- Google's `Search Central <https://developers.google.com/search/docs>`__
- Google's `How Search Works <https://www.google.com/search/howsearchworks/how-search-works/ranking-results/>`_ guide
- Bing's `Webmaster Guidelines <https://www.bing.com/webmasters/help/webmasters-guidelines-30fba23a>`__

WTD resources on SEO
~~~~~~~~~~~~~~~~~~~~

> - :doc:`Topics archive on search </topics>`

Reviewer: this works! yay!

Books
~~~~~

- `SEO for Everyone <https://abookapart.com/products/seo-for-everyone>`__, by Rebekah Baggs & Chris Corak

Blogs and websites
~~~~~~~~~~~~~~~~~~

- `Moz SEO Beginners Guide <https://moz.com/beginners-guide-to-seo>`__
- `The Ultimate Guide to SEO Meta Tags <https://moz.com/blog/the-ultimate-guide-to-seo-meta-tags>`__ by Moz
- `Search Engine Land <https://searchengineland.com/>`__
- `Hubspot's list of SEO blogs <https://blog.hubspot.com/marketing/best-seo-blogs>`__
- `Search Engine Optimization: A beginner’s guide from Ahrefs <https://ahrefs.com/seo>`__

> - `SEO Marketing Hub 2.0 from Backlinko <http://backlinko.com/hub/seo`__

Reviewer 2: Guess I was wrong. Maybe this one?
```suggestion
- `SEO Marketing Hub 2.0 from Backlinko <http://backlinko.com/hub/seo>`__
```

Author: Well, `<head>` was still wrong.
This one is too, fixing it.

Thanks again for your good eye.
